lib/GADS/API.pm

```diff
@@ -546,6 +546,27 @@ any ['get', 'post'] => '/api/:sheet/records' => require_login sub {
     _get_records();
 };
 
+get '/api/:sheet/fields' => require_login sub {
+    my $user             = logged_in_user;
+    my $sheetname        = param 'sheet'
+        or panic __"No sheet name found";
+    my $layout           = var('instances')->layout_by_shortname($sheetname)
+        or panic __x"Sheet {name} not found", name => $sheetname;
+    my $col_title        = query_parameters->get('title')
+        or panic __"No column title found";
+    my $search           = query_parameters->get('search');
+
+    my @columns = $layout->all(user_can_read => 1);
+    #Not sure if I should use the below in case it returns the column regardless of permissions?
+    #my $column = $layout->column_by_name($col_title);
```

Comment:
It would be better to use the column ID instead, as that is guaranteed to be unique. You could then do:
```
my $column = $layout->column($col_id, permission => "read")
    or error __x"Column {id} not found", id => $col_id;
```